src/seismometer/data/pandas_helpers.py

```diff
@@ -330,31 +369,40 @@ def _handle_merge(
         The column in the right (event) frame to use reference in the distance match, by default 'Time'.
     event_display : str, optional
         The name of the event to display in warning messages, by default "an event"
+    merge_strategy : str
+        The method to use when merging the event data, by default 'forward'.
 
     Returns
     -------
     pd.DataFrame
         The merged dataframe.
     """
-
-    ct_times = one_event[event_ref].notna().sum()
-    if ct_times == 0:
-        logger.warning(f"No times found for {event_display}, merging 'first'")
-        return pd.merge(predictions, one_event.groupby(pks).first(), how="left", on=pks)
-
-    if ct_times != len(one_event.index):
-        logger.warning(f"Inconsistent event times for {event_display}, only considering events with times.")
-        one_event = one_event.dropna(subset=[event_ref])
-
-    # merge next event for each prediction
-    return pd.merge_asof(
-        predictions,
-        one_event.dropna(subset=[event_ref]),
-        left_on=pred_ref,
-        right_on=event_ref,
-        by=pks,
-        direction="forward",
-    )
+    if merge_strategy=="forward" or merge_strategy=="nearest":
```

Comment:
Could you elaborate a bit? I'm not really certain what you mean here.

I do agree that "forward" and "nearest" are maybe a bit ambiguous but they're the exact values allowed by merge_asof's direction arg and they're passed in directly. 

This is an aside, but "backwards" is also a valid value for direction so maybe it should also be allowed here?

Reply:
I apologize, I'm still not certain I'm following exactly. 
If you're referencing "next" because of the code comment I agree that was confusing and inaccurate so I've changed it.
https://github.com/epic-open-source/seismometer/blob/4909413838d4a97c0ff81e3de0c5776d1d833642/src/seismometer/data/pandas_helpers.py#L120

"nearest" is just closest value, so that can be forward or backwards. In your example 0.999 would be closer to 1. "forward" is nearest after.